src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/config/prelude.hpp

```diff
@@ -89,22 +89,23 @@
 #undef BSONCXX_VERSION_PATCH
 
 // export.hpp (generated by CMake)
-#pragma push_macro("BSONCXX_API_H")
-#undef BSONCXX_API_H
+#pragma push_macro("BSONCXX_ABI_EXPORT_H")
+#if !defined(BSONCXX_ABI_EXPORT_H)
```

Comment:
It is not strictly needed for v_noabi headers, where all config headers are included via `prelude.hpp`.

This pattern is copied from the v1 export header, which permits direct inclusion of `export.hpp` without going through `prelude.hpp`. Because this pattern works generally, I elected to use it here now, as it will ultimately be moved into v1 headers anyways.

To explain the difference, current macro guard headers are designed to be used in the following manner:

```cpp
#include <bsoncxx/normal/headers.hpp>

// Comes AFTER all normal headers.
// Includes <bsoncxx/config/export.hpp> here.
#include <bsoncxx/config/prelude.hpp>

BSONCXX_ABI_EXPORT_CDECL(void) usage();

#include <bsoncxx/config/postlude.hpp>
```

The `push+if` of the include guard macro is designed to support direct inclusion of `<export.hpp>` rather than going through `prelude.hpp` in v1 headers:

```cpp
// Comes BEFORE all normal headers.
#include <bsoncxx/v1/prelude.hpp>

#include <bsoncxx/v1/config/export.hpp>
#include <bsoncxx/v1/normal/headers.hpp>

BSONCXX_ABI_EXPORT_CDECL(void) usage();

#include <bsoncxx/v1/detail/postlude.hpp>
```

while also remaining backward-compatible with the v_noabi `prelude.hpp` method of export header inclusion (in the v1 branch, v_noabi headers include their v1 counterparts):

```cpp
// May or may not include <bsoncxx/v1/config/export.hpp> directly.
#include <bsoncxx/normal/headers.hpp>

// Transitively includes:
//   -> <bsoncxx/v1/detail/prelude.hpp>
//   -  /* push_macro + undef */
//   -> <bsoncxx/config/export.hpp>
//     -> <bsoncxx/v1/detail/export.hpp>
#include <bsoncxx/config/prelude.hpp>

BSONCXX_ABI_EXPORT_CDECL(void) usage();

// Transitively includes:
//   -  /* pop_macro */
//   -> <bsoncxx/v1/detail/postlude.hpp>
#include <bsoncxx/config/postlude.hpp>
```

Reply:
Reverted to familiar macro guard pattern to avoid confusion with its interaction with the current state of headers.